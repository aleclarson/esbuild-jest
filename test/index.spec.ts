import mockfs = require('mock-fs')
import esbuildJest from '../src/index'

const { process } = esbuildJest.createTransformer({})

afterEach(() => {
  mockfs.restore()
})

test('ts file', () => {
  const content = `
    import names from './names'

    export function display() {
      return names
    }
  `
  const output = process(content, './index.ts')
  expect(output.code).toMatchSnapshot()
  expect(output.map).toMatchSnapshot()
})

test('tsx file', async () => {
  const content = `
    export default class Foo {
      render() {
        return <div className="hehe">hello there!!!</div>
      }
    }
  `
  const output = process(content, './index.tsx')
  expect(output.code).toMatchSnapshot()
  expect(output.map).toMatchSnapshot()
})
