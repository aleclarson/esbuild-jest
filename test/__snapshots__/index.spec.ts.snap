// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts file 1`] = `
"var __create = Object.create;
var __defProp = Object.defineProperty;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __markAsModule = (target) => __defProp(target, \\"__esModule\\", {value: true});
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, {get: all[name], enumerable: true});
};
var __exportStar = (target, module2, desc) => {
  if (module2 && typeof module2 === \\"object\\" || typeof module2 === \\"function\\") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== \\"default\\")
        __defProp(target, key, {get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable});
  }
  return target;
};
var __toModule = (module2) => {
  return __exportStar(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, \\"default\\", module2 && module2.__esModule && \\"default\\" in module2 ? {get: () => module2.default, enumerable: true} : {value: module2, enumerable: true})), module2);
};
__markAsModule(exports);
__export(exports, {
  display: () => display
});
var import_names = __toModule(require(\\"./names\\"));
function display() {
  return import_names.default;
}

//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4vaW5kZXgudHMiXSwic291cmNlc0NvbnRlbnQiOm51bGwsIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFDSSxtQkFBa0I7QUFFWCxtQkFBbUI7QUFDeEIsU0FBTztBQUFBOyIsIm5hbWVzIjpbXX0="
`;

exports[`ts file 2`] = `
Object {
  "mappings": ";;;;;;;;;;;;;;;;;;;;;;AAAA;AAAA;AAAA;AAAA;AACI,mBAAkB;AAEX,mBAAmB;AACxB,SAAO;AAAA;",
  "names": Array [],
  "sources": Array [
    "./index.ts",
  ],
  "sourcesContent": null,
  "version": 3,
}
`;

exports[`tsx file 1`] = `
"var __defProp = Object.defineProperty;
var __markAsModule = (target) => __defProp(target, \\"__esModule\\", {value: true});
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, {get: all[name], enumerable: true});
};
__markAsModule(exports);
__export(exports, {
  default: () => Foo
});
class Foo {
  render() {
    return /* @__PURE__ */ React.createElement(\\"div\\", {
      className: \\"hehe\\"
    }, \\"hello there!!!\\");
  }
}

//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4vaW5kZXgudHN4Il0sInNvdXJjZXNDb250ZW50IjpudWxsLCJtYXBwaW5ncyI6Ijs7Ozs7O0FBQUE7QUFBQTtBQUFBO0FBQUE7QUFDSSxVQUF5QjtBQUFBLEVBQ3ZCLFNBQVM7QUFDUCxXQUFPLG9DQUFDLE9BQUQ7QUFBQSxNQUFLLFdBQVU7QUFBQSxPQUFPO0FBQUE7QUFBQTsiLCJuYW1lcyI6W119"
`;

exports[`tsx file 2`] = `
Object {
  "mappings": ";;;;;;AAAA;AAAA;AAAA;AAAA;AACI,UAAyB;AAAA,EACvB,SAAS;AACP,WAAO,oCAAC,OAAD;AAAA,MAAK,WAAU;AAAA,OAAO;AAAA;AAAA;",
  "names": Array [],
  "sources": Array [
    "./index.tsx",
  ],
  "sourcesContent": null,
  "version": 3,
}
`;
